ql/src/java/org/apache/hadoop/hive/ql/exec/OrcFileMergeOperator.java

```diff
@@ -154,14 +161,6 @@ private void processKeyValuePairs(Object key, Object value)
         return;
       }
 
-      // next file in the path
```

Comment:
Are you using the same OrcReader for all the input files? Is that because all the orc files are homogeneous?

Reply:
yes all the orc files are compatible, this operator only needs to use one OrcReader to read out the schema, which is used to create a new Orc writer for the final file. afterwards it uses filesystem.read(file, offset, length) on each orc file to read the stripe and copy into the final file.